File: axel-erp-backend/src/main/java/com/axelerp/service/InventoryService.java
package com.axelerp.service;

import com.axelerp.exception.ResourceNotFoundException;
import com.axelerp.model.Product;
import com.axelerp.model.PurchaseOrder;
import com.axelerp.model.Supplier;
import com.axelerp.repository.ProductRepository;
import com.axelerp.repository.PurchaseOrderRepository;
import com.axelerp.repository.SupplierRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class InventoryService {
    
    @Autowired
    private ProductRepository productRepository;
    
    @Autowired
    private SupplierRepository supplierRepository;
    
    @Autowired
    private PurchaseOrderRepository purchaseOrderRepository;
    
    // Product methods
    public List<Product> getAllProducts() {
        return productRepository.findAll();
    }
    
    public Product getProductById(Long id) {
        return productRepository.findById(id)
                .orElseThrow(() -> new ResourceNotFoundException("Product", "id", id));
    }
    
    public Product createProduct(Product product) {
        return productRepository.save(product);
    }
    
    public Product updateProduct(Long id, Product productDetails) {
        Product product = getProductById(id);
        product.setName(productDetails.getName());
        product.setDescription(productDetails.getDescription());
        product.setSku(productDetails.getSku());
        product.setPrice(productDetails.getPrice());
        product.setQuantity(productDetails.getQuantity());
        product.setReorderLevel(productDetails.getReorderLevel());
        product.setCategory(productDetails.getCategory());
        return productRepository.save(product);
    }
    
    public void deleteProduct(Long id) {
        Product product = getProductById(id);
        productRepository.delete(product);
    }
    
    public List<Product> getLowStockProducts() {
        return productRepository.findAll().stream()
                .filter(p -> p.getQuantity() <= p.getReorderLevel())
                .toList();
    }
    
    // Supplier methods
    public List<Supplier> getAllSuppliers() {
        return supplierRepository.findAll();
    }
    
    public Supplier createSupplier(Supplier supplier) {
        return supplierRepository.save(supplier);
    }
    
    // Purchase Order methods
    public List<PurchaseOrder> getAllPurchaseOrders() {
        return purchaseOrderRepository.findAll();
    }
    
    public PurchaseOrder createPurchaseOrder(PurchaseOrder purchaseOrder) {
        return purchaseOrderRepository.save(purchaseOrder);
    }
}

File: axel-erp-frontend/src/services/inventoryService.js
import api from './api';

const inventoryService = {
  // Products
  getAllProducts: () => api.get('/inventory/products'),
  getProductById: (id) => api.get(`/inventory/products/${id}`),
  createProduct: (product) => api.post('/inventory/products', product),
  updateProduct: (id, product) => api.put(`/inventory/products/${id}`, product),
  deleteProduct: (id) => api.delete(`/inventory/products/${id}`),
  getLowStockProducts: () => api.get('/inventory/products/low-stock'),

  // Suppliers
  getAllSuppliers: () => api.get('/inventory/suppliers'),
  createSupplier: (supplier) => api.post('/inventory/suppliers', supplier),

  // Purchase Orders
  getAllPurchaseOrders: () => api.get('/inventory/purchase-orders'),
  createPurchaseOrder: (order) => api.post('/inventory/purchase-orders', order),
};

export default inventoryService;
